apps/src/applab/firebaseStorage.js

```diff
@@ -398,4 +400,82 @@ FirebaseStorage.renameColumn = function (tableName, oldName, newName, onSuccess,
     .then(onSuccess, onError);
 };
 
+/**
+ * Parses a CSV string into records data in a format that can be written into the
+ * records node of a firebase table.
+ * @param {string} csvData
+ * @returns {Promise} A promise containing recordsData or an error message.
+ */
+function parseRecordsDataFromCsv(csvData) {
+  return new Promise((resolve, reject) => {
+    parseCsv(csvData, {columns: true}, (error, records) => {
+      if (error) {
+        return reject(error);
+      }
+      resolve(records);
+    });
+  }).then(records => {
+    let recordsData = {};
+    records.forEach((record, index) => {
+      const id = index + 1;
+      Object.keys(record).forEach(key => {
+        record[key] = castValue(record[key]);
+      });
+      record.id = id;
+      recordsData[id] = JSON.stringify(record);
+    });
+    return recordsData;
+  });
+}
+
+/**
+ * Validates that the records data does not exceed size limits.
+ * @param recordsData The records data to validate.
+ * @returns {Promise} A promise containing recordsData or an error message.
+ */
+function validateRecordsData(recordsData) {
+  return loadConfig().then(config => {
+    if (Object.keys(recordsData).length > config.maxTableRows) {
+      return Promise.reject(`Import failed because the data is too large. ` +
+        `A table may only contain ${config.maxTableRows} rows.`);
+    }
+    if (Object.keys(recordsData).some(record => record.length > config.maxRecordSize)) {
```

Comment:
Good catch! This is in fact a bug. I forgot to test this (and clearly need to add real tests soon).